### PR TITLE
[CONTP-395] Remove dependency of orchestrator cluster check on the tagger for unassigned pods tagging

### PR DESCRIPTION
**Pull request by adel121**
### What does this PR do?

This PR removes the dependency of orchestrator cluster check on the tagger for unassigned pods tagging. 

After this PR, the orchestrator cluster check will be calculating the tags of unassigned pods each time on demand by using the pod tag provider, without needing to populate the tagger with pod tags.

### Motivation

On large clusters, when the orchestrator cluster check is using the remote cluster tagger, it receives a large amount of data describing tags of all pods on the cluster.

However, the orchestrator cluster check running in a cluster check runner is only interested in unassigned pods in the cluster (i.e. pods that are not yet scheduled on a node). 

Currently, there is no way to subscribe to only unassigned pods with the remote tagger.

For this reason, this PR is helpful to not require syncing pod tags with the cluster tagger, and still be able to attach tags to unassigned pods.

### Describe how to test/QA your changes

Normally, E2E should verify the correctness of this change. 

But it is still good to do a second check because it is a sensitive change:

## Verify it works in CLC runner:

Create an unschedulable pod on a kubernetes cluster:

You can use this template:
```
apiVersion: v1
kind: Pod
metadata:
  name: unschedulable-pod-sample
  labels:
    pod-team: contp
  annotations:
    pod-domain: research
spec:
  containers:
    - name: nginx
      image: nginx
  nodeSelector:
    nonExistentLabel: "true"
```

Deploy the agent, dca and clc runner with orchestrator cluster check config:

```
datadog:
  apiKeyExistingSecret: datadog-secret
  appKeyExistingSecret: datadog-secret
  logLevel: DEBUG
  clusterName: adel-orchestrator-clc-1
  clusterChecks:
    enabled: true
  kubelet:
    tlsVerify: false

  podLabelsAsTags:
    pod-team: team

  podAnnotationsAsTags:
    pod-domain: domain

clusterChecksRunner:
  enabled: true

  env:
    - name: DD_CLC_RUNNER_REMOTE_TAGGER_ENABLED
      value: true
    # We need to manually set pod labels and annotations as tags in CLC runner because dd helm doesn't automatically pass the config down to the clc runner. This will be updated in a separate PR.
    - name: DD_KUBERNETES_POD_LABELS_AS_TAGS
      value: '{"pod-team": "team"}'
    - name: DD_KUBERNETES_POD_ANNOTATIONS_AS_TAGS
      value: '{"pod-domain": "domain"}'

clusterAgent:
  enabled: true
  replicas: 1

  advancedConfd:
    orchestrator.d:
      1.yaml: |-
        cluster_check: true
        init_config:
        instances:
          - collectors:
            - pods
            skip_leader_election: true
```

For this to work, the CLC runner should have the permission to watch pods, which is not properly handled in the helm chart, so you should update the rbacs of the CLC runner. The easiest and fastest shortcut to do this is to copy the rbacs from the DCA to the CLC runner.

You should do the following:
Copy the list of rules from the dca clusterrole.
You can get the dca clusterrole by running the following command
```
kubectl get clusterrole datadog-agent-cluster-agent -o yaml
```

Append the copied rules to the clusterrole of the agent:
```
kubectl edit clusterrole datadog-agent
```

After some time, you should be able to see your pods on the kubernetes explorer page in the DD UI:
image

You should be able to find the following labels on the unschedulable-pod-sample:
- `team:contp`
- `domain:research`

## Verify it works in DCA:

Repeat the same process as in the previous section, but now set `cluster_check: false` in the orchestrator check config to let the check run in the DCA. 

In this case, no need for copying rbac rules, and you should get the same results at the end.

### CODE REVIEW

**Review by vickenty — APPROVED** [review]

File: pkg/collector/corechecks/cluster/orchestrator/processors/k8s/pod_tag_provider/provider.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+type PodTagProvider interface {
+	GetTags(*corev1.Pod, taggertypes.TagCardinality) ([]string, error)
+}
+
+// NewPodTagProvider returns a new PodTagProvider
+// In case of CLC runner, the provider will calculate tags based on the pod resource on demand each time
+// In case of Node agent or Cluster agent, the provider will get pod tags by querying the tagger
+func NewPodTagProvider(cfg config.Component, store workloadmeta.Component) PodTagProvider {
+	if flavor.GetFlavor() != flavor.ClusterAgent && pkgconfigsetup.IsCLCRunner(pkgconfigsetup.Datadog()) {
+		// Running in a CLC Runner
```
Comment (davidor) [inline pkg/collector/corechecks/cluster/orchestrator/processors/k8s/pod_tag_provider/provider.go:32]: Isn't the first condition of the `if` redundant if this is only about CLC?

Comment (adel121) [reply pkg/collector/corechecks/cluster/orchestrator/processors/k8s/pod_tag_provider/provider.go:32]: Actually here I want to distinguish between two cases:
- running the check in a node agent
- running the check in a cluster check runner

Checking `isCLCRunner()` alone is not enough to make the distinction.

But you are still correct that the first condition is redundant. However, I think the statement should be changed to this:

```
if flavor.GetFlavor() != flavor.ClusterAgent && !pkgconfigsetup.IsCLCRunner(pkgconfigsetup.Datadog()) {
// this means we are in the orchestrator check version that runs in the node agent
return newNodePodTagProvider()
}

return newCLCTagProvider(cfg, store)
```

WDYT?

**Review by xlucas — APPROVED** [review]
LGTM

**Review by davidor — APPROVED** [review]

**Pull request merged**